src/domain/opendatafabric/src/identity/dataset_id.rs

```diff
@@ -204,3 +204,56 @@ impl<'de> serde::de::Visitor<'de> for DatasetIDSerdeVisitor {
         DatasetID::from_did_str(v).map_err(serde::de::Error::custom)
     }
 }
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/// SQLX
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[cfg(feature = "sqlx")]
+impl<'r, DB: sqlx::Database> sqlx::Decode<'r, DB> for DatasetID
+where
+    &'r str: sqlx::Decode<'r, DB>,
```

Comment:
I guess this part was copypasted from `AccountID`. Should we have a macro for such newtypes?